R/derive_interp_records.R

```diff
@@ -0,0 +1,105 @@
+#' Derive interpolated rows for the CDC charts (>=2 yrs old)
+#'
+#' Derive interpolated rows for the CDC charts (>=2 yrs old) by age in days
+#' for the following parameters: HEIGHT, WEIGHT and BMI
+#'
+#' @param dataset Input metadataset
+#'
+#'   The variables `AGE`, `AGEU`, `SEX`, `L`, `M`, `S` are expected to be in the dataset
+#'
+#'   For BMI the additional variables `P95` and `Sigma` are expected to be in the dataset
+#'
+#'   Note that `AGE` must be in days so that `AGEU` is equal to `"DAYS"`
+#'
+#' @param by_vars Grouping variables
+#'
+#'   The variable from `dataset` which identifies the group of observations
+#'   to interpolate separately.
+#'
+#' @param parameter CDC/WHO metadata parameter
+#'
+#' *Permitted Values*: `"WEIGHT"`, `"HEIGHT"` or `"BMI"` only - Must not be `NULL`
+#'
+#'   e.g. `parameter = "WEIGHT"`,
+#'        `parameter = "HEIGHT"`,
+#'   or   `parameter = "BMI"`.
+#'
+#' @return The input dataset plus additional interpolated records.
+#'
+#' @family metadata
+#'
+#' @keywords metadata
+#'
+#' @export
+#'
+#' @examples
+#' library(dplyr, warn.conflicts = FALSE)
+#'
+#' cdc_htage <- admiralpeds::cdc_htage %>%
+#'   mutate(
+#'     SEX = case_when(
+#'       SEX == 1 ~ "M",
+#'       SEX == 2 ~ "F",
+#'       TRUE ~ NA_character_
+#'     ),
+#'     # Ensure first that Age unit is "DAYS"
+#'     AGE = round(AGE * 30.4375)
+#'   ) %>%
+#'   # Interpolate the AGE by SEX
+#'   derive_interp_records(
+#'     by_vars = exprs(SEX),
+#'     parameter = "HEIGHT"
+#'   )
+#'
+#' print(cdc_htage)
+derive_interp_records <- function(dataset,
+                                  by_vars = NULL,
+                                  parameter) {
+  # Apply assertions to each argument to ensure each object is appropriate class
+  by_vars <- assert_vars(by_vars, optional = TRUE)
+  assert_character_scalar(parameter, values = c("HEIGHT", "WEIGHT", "BMI"))
+  assert_data_frame(dataset, required_vars = exprs(!!!by_vars, AGE, L, M, S))
+  if (parameter == "BMI") {
+    assert_data_frame(dataset, required_vars = exprs(!!!by_vars, AGE, L, M, S, P95, Sigma))
```

Comment:
in the above 2 `assert_data_frame()` calls we have `!!!by_vars` as required, but then `by_vars=NULL` is allowed in this function

Reply:
it works fine while `by_vars = NULL`: Tested in the ADVS template

Reply:
This shouldn't matter. If `by_vars = NULL`, `exprs(!!!by_vars, AGE, L, M, S)` is the same as `exprs(AGE, L, M, S))`. I.e., `NULL` arguments are ignored.

Reply:
ah ok sorry, my lack of understanding! sorry @Fanny-Gautier - i just wanted to check that these still worked for `by_vars=NULL`, that was my main comment. please do put these back to your simpler way of having this.